app/ts/library/utilities.ts

```diff
@@ -42,3 +42,17 @@ export function isEthereumObservable(ethereum: unknown): ethereum is EthereumWit
 export function assertUnreachable(value: never): never {
 	throw new Error(`Never gonna give you up (${value})`)
 }
+
+export function isTransactionHash(hash: string): hash is `0x${string}` {
+	return ethers.utils.hexDataLength(hash) === 32
+}
+
+export function assertsTransactionHash(hash: string): asserts hash is `0x${string}` {
+	if (!isTransactionHash(hash)) throw new Error('Invalid transaction hash')
+}
+
+export const calculateGasFee = (effectiveGasPrice: ethers.BigNumber, gasUsed: ethers.BigNumber) => {
+	const gasFeeBigNum = effectiveGasPrice.mul(gasUsed)
+	const gasFee = ethers.utils.formatEther(gasFeeBigNum)
+	return gasFee
+}
```

Comment:
Not really an issue at the moment, but at some point you may want to split out the Ethereum specific utilities from the more generic utilities (like sleep, removeNonStringsAndTrim).